### PR TITLE
Validate Phone Number

### PR DESCRIPTION
**Pull request by sampoder**
Fixes #3982

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -312,8 +312,10 @@ def update
 
     if @user.update(user_params)
       confetti! if !@user.seasonal_themes_enabled_before_last_save && @user.seasonal_themes_enabled? # confetti if the user enables seasonal themes
-
-      if @user.full_name_before_last_save.blank?
+      if !((user_params[:phone_number] =~ /\A\+[1-9]\d{1,14}\z/)) && user_params[:phone_number]
```
Comment (cjdenio) [inline app/controllers/users_controller.rb]: Could you move this to a [model-level validation](https://guides.rubyonrails.org/active_record_validations.html)?

**Review by cjdenio — CHANGES_REQUESTED** [review]

**Comment by sampoder** [comment]
oh awesome @cjdenio, TIL! just updated, let me know if i did smth wrong :D

**Comment by sampoder** [comment]
Seems like I've messed something up..

**Comment by cjdenio** [comment]
@sampoder it looks like we're already using [this gem](https://github.com/daddyz/phonelib) for phone validation— I wonder if it's configured incorrectly

**Comment by northeastprince** [comment]
Seems to be validating phone numbers correctly already on my end.

**Comment by cjdenio** [comment]
Has that changed since [this comment](https://github.com/hackclub/hcb/issues/3982#issuecomment-1593549955)?

**Comment by northeastprince** [comment]
Hmm, maybe the validations weren't done for some reason on that one or we need to format numbers in E164?

**Comment by sampoder** [comment]
@northeastprince what was the phone number you used that broke it?

**Comment by sampoder** [comment]
I guess I'll close this for now

**Pull request closed without merging**